### PR TITLE
Converts oil for borgs into a status effect and rebalances it

### PR DESCRIPTION
**Pull request by ThatFiveGuys**
## About the PR 
If you want the tldr of the new oil behavior, in short being oiled recently makes you freshly oiled which makes you go really fast and power efficent for a max of 15 seconds. This decays into the oiled status effect for a max of 6 minutes, which makes you go go somewhat fast and somewhat power efficent.  Just look at the video below
https://github.com/goonstation/goonstation/assets/73148980/0258ad07-86d7-4330-8d11-113844f35392

To sum up the old oil behavior for probably unneeded context, one unit of oil applied 2 seconds of being oiled, this status effect reduces your speed by .5, reduces your max stun by 25% + stun resistance by 25, and halves your passive power consumption. This had no clamp.

To sum up the new behavior, this pr splits oil into two status effects, oiled and freshly oiled. Both status effects apply the stun modifiers described above. 

Every unit of oil applied to a borg applies half a second of the freshly oiled status effect. This status effect caps out at 15 seconds, meaning you need to apply 30 units to cap this effect. This halves your passive power consumption, and reduces your movement delay by 50%. When freshly oiled is finished, you get applied the oiled status effect with 24 times the duration you've had the freshly oiled status effect for, meaning you have to have freshly oiled for at least 15 seconds to cap out at 6 minutes.

Oiled status effect just reduces your movement delay and passive power consumption by 15%, and also does the same stun modifiers mentioned above.

## Why's this needed? 
This new behavior is both far more readable, and also far more interesting and fun, also hopefully well balanced.

## Changelog 

```changelog
(u)Ikea
(*)The previous behavior for oil in regards to borgs has been split into two status effects, freshly oiled and oiled. These apply speed increase and power consumption decrease in varying amounts, alongside stun resistance.
```

### CODE REVIEW

**Comment by ZeWaka** [comment]
gib bideo u posted in imcoder

**Comment by TDHooligan** [comment]
notable it nerfs light borgs' speed bonus more than it nerfs heavy borgs. i like this more than the existing effect

File: code/modules/status_system/statusEffects.dm [hunk 1]
```diff
@@ -2400,6 +2400,57 @@
...
+
+/datum/statusEffect/oiled/fresh
+	id = "freshly_oiled"
+	name = "Freshly oiled"
+	icon_state = "fresh_oil"
+	maxDuration = 15 SECONDS
+	movement_modifier = /datum/movement_modifier/robot_oil_fresh
+	var/oiledDuration = 0 // duration of the oiled status effect a person has before more oil is applied.
+	var/maxFresh = 0 // What is the max duration freshly oiled status has reached
+	var/tickspassed = 0 // How long have we had the status effect for
```
Comment (ZeWaka) [inline code/modules/status_system/statusEffects.dm:2433]:
```suggestion
	/// Duration of the oiled status effect a person has before more oil is applied.
	var/oiledDuration = 0
	/// What is the max duration freshly oiled status has reached
	var/maxFresh = 0 
	/// How long have we had the status effect for
	var/tickspassed = 0 
```

Comment (ZeWaka) [reply code/modules/status_system/statusEffects.dm:2433]: maxFresh is also unused?

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Review by ZeWaka — APPROVED** [review]

File: code/mob/living/silicon/robot.dm [hunk 2]
```diff
@@ -2508,7 +2512,10 @@
 					var/delta = src.max_upgrades - initial(src.max_upgrades)
 					power_use_tally += 3 ** delta
 
-				if (src.oil && power_use_tally > 0) power_use_tally /= 1.5
+				if (src.hasStatus("freshly_oiled") && power_use_tally > 0)
+					power_use_tally *= 0.5
+				if (src.hasStatus("oiled") && power_use_tally > 0)
+					power_use_tally *= 0.75
```
Comment (Tarmunora) [inline code/mob/living/silicon/robot.dm:2518]: inconsistent with the above

**Review by Tarmunora — CHANGES_REQUESTED** [review]
at 15 seconds, the "freshly oiled" thing seems pretty pointless

**Comment by ZeWaka** [comment]
> > at 15 seconds, the "freshly oiled" thing seems pretty pointless
> 
> Freshly oiled is mostly intended to be a short speed bonus for borgs to apply on themselves sorta similar to how current behavior of oil can be utilized. Also borgs going mach 12 funny.

Yeah, I also thought of/requested/proposed this because it adds a bit of complexity to an otherwise boring "set and forget" feature

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**